src/Core/Configurations/RuntimeConfigValidator.cs

```diff
@@ -830,11 +830,161 @@ public void ValidateRelationshipsInConfig(RuntimeConfig runtimeConfig, IMetadata
+                // from the source entity to the linkingObject while target and linkingTarget fields provide the relationship from the target entity to the
+                // linkingObject.
+                // see: https://learn.microsoft.com/en-us/azure/data-api-builder/relationships#many-to-many-relationship
+                if (relationship.LinkingObject is not null)
+                {
+                    if (relationship.SourceFields is not null && relationship.LinkingSourceFields is null)
+                    {
+                        // Validation to ensure that if source fields exist that linking source fields exist as well.
+                        HandleOrRecordException(new DataApiBuilderException(
+                            message: $"Entity: {entityName} has a many to many relationship: {relationshipName} with source fields that are not null, " +
+                                $"but linking source fields that are null.",
+                            statusCode: HttpStatusCode.ServiceUnavailable,
+                            subStatusCode: DataApiBuilderException.SubStatusCodes.ConfigValidationError));
+                    }
+
+                    if (relationship.LinkingSourceFields is not null && relationship.SourceFields is null)
+                    {
+                        // Validation to ensure that if linking source fields exist that source fields exist as well.
+                        HandleOrRecordException(new DataApiBuilderException(
+                            message: $"Entity: {entityName} has a many to many relationship: {relationshipName} with linking source fields that are not null, " +
+                                $"but source fields that are null.",
+                            statusCode: HttpStatusCode.ServiceUnavailable,
+                            subStatusCode: DataApiBuilderException.SubStatusCodes.ConfigValidationError));
+                    }
+
+                    if (relationship.TargetFields is not null && relationship.LinkingTargetFields is null)
+                    {
+                        // Validation to ensure that if target fields exist that linking target fields exist as well.
+                        HandleOrRecordException(new DataApiBuilderException(
+                            message: $"Entity: {entityName} has a many to many relationship: {relationshipName} with target fields that are not null, " +
+                                $"but linking target fields that are null.",
+                            statusCode: HttpStatusCode.ServiceUnavailable,
+                            subStatusCode: DataApiBuilderException.SubStatusCodes.ConfigValidationError));
+                    }
+
+                    // Validation to ensure that if linking target fields exist that targets fields exist as well.
+                    if (relationship.LinkingTargetFields is not null && relationship.TargetFields is null)
```

Comment:
All these validations are between 2 pairs of fields:
1) SourceFields, TargetFields
2) SourceFields, LinkingSourceFields
3) TargetFields, LinkingTargetFields 

- there can easily be defined an helper function that takes a pair to validate. That helper can be called 3 times to reduce code duplication.